src/main/host/syscall/handler/socket.rs

```diff
@@ -234,31 +248,35 @@ impl SyscallHandler {
         let addr_ptr: PluginPtr = args.get(4).into();
         let addr_len_ptr: PluginPtr = args.get(5).into();
 
-        // get the descriptor, or return early if it doesn't exist
-        let desc = match Self::get_descriptor(ctx.process, fd)? {
-            CompatDescriptor::New(desc) => desc,
-            // if it's a legacy descriptor, use the C syscall handler instead
-            CompatDescriptor::Legacy(_) => {
-                return unsafe {
-                    c::syscallhandler_recvfrom(
-                        ctx.thread.csyscallhandler(),
-                        args as *const c::SysCallArgs,
-                    )
-                    .into()
+        // if we were previously blocked, get the active file from the last syscall handler
+        // invocation since it may no longer exist in the descriptor table
+        let file = ctx
+            .thread
+            .syscall_condition()
+            // if this was for a C descriptor, then there won't be an active file object
+            .map(|x| x.active_file().cloned())
+            .flatten();
+
+        let file = match file {
```

Comment:
optional nit - maybe slightly more compact and idiomatic to do something like:

```
let file = file.unwrap_or_else(|| {
...
});
```

Reply:
The issue with this is that there's a short-circuit path (`return unsafe { c::syscallhandler_sendto(...) }`) if it's a legacy descriptor, so if you put this in a closure, then you only short-circuit the closure and not the syscall handler.

Reply:
Ah, gotcha. Nevermind then